ksql-engine/src/test/java/io/confluent/ksql/planner/plan/JoinNodeTest.java

```diff
@@ -238,4 +302,619 @@ public void shouldHaveAllFieldsFromJoinedInputs() {
     final Set<String> fields = stream.getSchema().fields().stream().map(Field::name).collect(Collectors.toSet());
     assertThat(fields, equalTo(expected));
   }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldPerformStreamToStreamLeftJoin() {
+    setupStream(left, leftSchemaKStream, leftSchema, 2);
+    expectKeyField(leftSchemaKStream, leftKeyFieldName);
+
+    setupStream(right, rightSchemaKStream, rightSchema, 2);
+
+    final Field joinKey = joinSchema.field(leftAlias + "." + leftKeyFieldName);
+
+    final SpanExpression spanExpression = new SpanExpression(10, TimeUnit.SECONDS);
+
+    expect(leftSchemaKStream.leftJoin(eq(rightSchemaKStream),
+                                      eq(joinSchema),
+                                      eq(joinKey),
+                                      eq(spanExpression.joinWindow()),
+                                      anyObject(Serde.class),
+                                      anyObject(Serde.class)))
+        .andReturn(niceMock(SchemaKStream.class));
+
+    replay(left, right, leftSchemaKStream, rightSchemaKStream);
+
+    final JoinNode joinNode = new JoinNode(new PlanNodeId("join"),
+                                           JoinNode.JoinType.LEFT,
+                                           left,
+                                           right,
+                                           leftKeyFieldName,
+                                           rightKeyFieldName,
+                                           leftAlias,
+                                           rightAlias,
+                                           spanExpression,
+                                           DataSource.DataSourceType.KSTREAM,
+                                           DataSource.DataSourceType.KSTREAM);
+
+    joinNode.buildStream(mockStreamsBuilder,
+                         mockKsqlConfig,
+                         mockKafkaTopicClient,
+                         mockFunctionRegistry,
+                         properties,
+                         mockSchemaRegistryClient);
+
+    verify(left, right, leftSchemaKStream, rightSchemaKStream);
+
+    assertEquals(leftKeyFieldName, joinNode.getLeftKeyFieldName());
```

Comment:
Why have all these checks? These fields are just set by the constructor.

Reply:
That's the behavior now. But we want to make sure that if it changes, at least we catch it and do it consciously. No harm in checking IMO.